src/optimagic/optimization/internal_optimization_problem.py

```diff
@@ -704,6 +717,33 @@ def _pure_evaluate_fun_and_jac(
 
         return (algo_fun_value, out_jac), hist_entry, log_entry
 
+    def _check_infinite_gradients(
+        self,
+        params: PyTree,
+        out_jac: NDArray[np.float64],
+        jac_value: PyTree,
+    ) -> None:
+        """Check for infinite values in gradients and raise an error if found.
+
+        Args:
+            x: internal parameter vector at which the gradient was evaluated.
+            params: user-facing parameter representation at evaluation point.
+            out_jac: internal processed gradient to check for infinities.
+            jac_value: original gradient value as returned by the user function,
+                    included in error messages for debugging.
+
+        Raises:
+            InvalidFunctionError: If any infinite values are found in the gradient.
+
+        """
+        if np.any(np.isinf(out_jac)) or np.any(np.isnan(out_jac)):
```

Comment:
Given the adjusted function name, I would also use

```python
if not np.all(np.isfinite(out_jac))
...
```

here. For this you should then adjust the docstring and add a comment in the docstring that `np.isfinite` checks for infinity and NaN values (see [`np.isfinite` docstring](https://numpy.org/doc/2.2/reference/generated/numpy.isfinite.html)).